lib/ecto/adapters/sql.ex

```diff
@@ -141,11 +146,17 @@ defmodule Ecto.Adapters.SQL do
   def to_sql(kind, repo, queryable) do
     adapter = repo.__adapter__
 
-    {_meta, prepared, params} =
+    plan =
       Ecto.Queryable.to_query(queryable)
       |> Ecto.Query.Planner.query(kind, repo, adapter)
-
-    {prepared, params}
+    case plan do
+      {:execute, _meta, %{statement: statement}, params} ->
```

Comment:
All of these sounds better. Yes the double meaning is awkward and I did try an approach without but I wanted to have minimal impact on the API. Shall I rework this to use unique integers and then you can refactor to a better API? The unique integers may impact the decision.